pkg/template/controller/readiness.go

```diff
@@ -0,0 +1,139 @@
+package controller
+
+import (
+	"strconv"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/apis/apps"
+	"k8s.io/kubernetes/pkg/apis/batch"
+	"k8s.io/kubernetes/pkg/apis/extensions"
+	deploymentutil "k8s.io/kubernetes/pkg/controller/deployment/util"
+
+	buildapi "github.com/openshift/origin/pkg/build/apis/build"
+	buildutil "github.com/openshift/origin/pkg/build/util"
+	"github.com/openshift/origin/pkg/client"
+	deployapi "github.com/openshift/origin/pkg/deploy/apis/apps"
+)
+
+func checkBuildReadiness(oc client.Interface, obj runtime.Object) (bool, bool, error) {
```

Comment:
i'd like to see a follow up PR that adds godoc(summarizing what is checked for the object)+unit tests to all these check functions.